src/main/java/app/linguistai/bmvp/service/gamification/LeaderboardService.java

```diff
@@ -0,0 +1,120 @@
+package app.linguistai.bmvp.service.gamification;
+
+import app.linguistai.bmvp.exception.NotFoundException;
+import app.linguistai.bmvp.exception.SomethingWentWrongException;
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.gamification.UserXP;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import app.linguistai.bmvp.repository.gamification.IUserXPRepository;
+import app.linguistai.bmvp.response.gamification.RLeaderboardXP;
+import app.linguistai.bmvp.response.gamification.RUserXPRanking;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class LeaderboardService {
+    private final IAccountRepository accountRepository;
+    private final IUserXPRepository xpRepository;
+
+    @Transactional
+    public RLeaderboardXP getTopUsersByExperience(String email, Integer page, int size) throws Exception {
+        try {
+            Long loggedUserRanking = xpRepository.findGlobalUserRankByEmail(email);
+            if (loggedUserRanking == null) {
+                throw new NotFoundException("User's global XP ranking", true);
+            }
+            // If page number is not given, then return the page where the user is present
+            if (page == null) {
+                page = getPageNumberByEmail(loggedUserRanking, size);
+            }
+
+            Pageable pageable = PageRequest.of(page, size, Sort.by("experience").descending());
```

Comment:
Making `"experience"` hard coded string a `constant` can help the maintainability of the code